src/ImageSharp/MetaData/Profiles/Exif/ExifReader.cs

```diff
@@ -15,38 +19,37 @@ namespace SixLabors.ImageSharp.MetaData.Profiles.Exif
     /// </summary>
     internal sealed class ExifReader
     {
-        private readonly Collection<ExifTag> invalidTags = new Collection<ExifTag>();
-        private byte[] exifData;
-        private uint currentIndex;
-        private bool isLittleEndian;
+        private readonly List<ExifTag> invalidTags = new List<ExifTag>();
+        private readonly byte[] exifData;
+        private int position;
+        private Endianness endianness = Endianness.BigEndian;
         private uint exifOffset;
         private uint gpsOffset;
-        private uint startIndex;
+        private int startIndex;
 
-        private delegate TDataType ConverterMethod<TDataType>(byte[] data);
+        public ExifReader(byte[] exifData)
+        {
+            DebugGuard.NotNull(exifData, nameof(exifData));
+
+            this.exifData = exifData;
+        }
+
+        private delegate TDataType ConverterMethod<TDataType>(ReadOnlySpan<byte> data);
 
         /// <summary>
         /// Gets the invalid tags.
         /// </summary>
-        public IEnumerable<ExifTag> InvalidTags => this.invalidTags;
+        public IList<ExifTag> InvalidTags => this.invalidTags;
```

Comment:
Is this list intended to be mutable from the outside?

Reply:
Nope. Changing to ReadOnly.